src/database/migration/file/migration_version_6.ts

```diff
@@ -0,0 +1,45 @@
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+import ChannelList from '../../schema/ChannelListSchema';
+import ChatSchema from '../../schema/ChatSchema';
+import Migration from './migration.types';
+
+const TABLE_NAME = ChatSchema.getTableName();
+
+class MigrationVersion6 implements Migration {
+  up = async (db: SQLiteDatabase): Promise<void> => {
+    const upQueryStep1 = `CREATE TABLE IF NOT EXISTS ${TABLE_NAME}_temp (
+        id UUID PRIMARY KEY NOT NULL,
+        channel_id TEXT NOT NULL,
+        user_id UUID NOT NULL,
+        message TEXT NOT NULL,
+        type TEXT NOT NULL CHECK(type IN ('regular', 'system')),
+        status TEXT NOT NULL CHECK(status IN ('pending', 'sent', 'delivered', 'read')),
+        created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        raw_json TEXT,
+        FOREIGN KEY (channel_id) REFERENCES ${ChannelList.getTableName()} (id),
+        FOREIGN KEY (user_id) REFERENCES users (id)
+      );`;
+
+    const upQueryStep2 = `INSERT INTO ${TABLE_NAME}_temp SELECT * FROM ${TABLE_NAME};`;
+    const upQueryStep3 = `DROP TABLE ${TABLE_NAME};`;
+    const upQueryStep4 = `ALTER TABLE ${TABLE_NAME}_temp RENAME TO ${TABLE_NAME};`;
+
+    console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    await db.executeSql(upQueryStep1);
+    await db.executeSql(upQueryStep2);
+    await db.executeSql(upQueryStep3);
+    await db.executeSql(upQueryStep4);
+    console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+
+  down = async (db: SQLiteDatabase): Promise<void> => {
+    console.log(`===== DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    const downQuery = `DROP TABLE IF EXISTS ${TABLE_NAME}`;
+    await db.executeSql(downQuery);
+    console.log(`===== DONE DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+}
+
+export default new MigrationVersion6();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The migration script seems to be well written. However, it's important to note that the `console.log` statements on lines 29, 34, 38, and 41 might not be necessary in a production environment. They could potentially expose sensitive information about your database structure. Consider removing them or replacing them with a more generic logging mechanism that doesn't reveal specific table names.

Also, there is no error handling around the `executeSql` calls. If any of these operations fail, the function will throw an unhandled promise rejection. It would be better to wrap these calls in a try-catch block to handle potential errors gracefully.

```diff
    console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
    try {
      await db.executeSql(upQueryStep1);
      await db.executeSql(upQueryStep2);
      await db.executeSql(upQueryStep3);
      await db.executeSql(upQueryStep4);
    } catch (error) {
      console.error(`Error migrating ${TABLE_NAME}:`, error);
      throw error;
    }
    console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
```

And similarly for the `down` method:

```diff
    console.log(`===== DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
    try {
      await db.executeSql(downQuery);
    } catch (error) {
      console.error(`Error downgrading ${TABLE_NAME}:`, error);
      throw error;
    }
    console.log(`===== DONE DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
```
This way, you can log the error and re-throw it to be handled by the caller.


<!-- This is an auto-generated comment by OSS CodeRabbit -->